mesonbuild/envconfig.py

```diff
@@ -80,7 +80,12 @@ def from_config_parser(cls, parser: configparser.ConfigParser) -> typing.Dict[st
         for s in parser.sections():
             section = {}
             for entry in parser[s]:
-                value = parser[s][entry]
+                if entry in parser.defaults():
+                    continue
+                try:
+                    value = parser[s][entry]
+                except configparser.InterpolationMissingOptionError as e:
+                    raise EnvironmentException(e.message)
```

Comment:
__[Flake8]__ [B306] `BaseException.message` has been deprecated as of Python 2.6 and is removed in Python 3. Use `str(e)` to access the user-readable message. Use `e.args` to access arguments passed to the exception. ([link](https://sider.review/gh/repos/19784232/pulls/5859#issue-4491595))

<sub>posted by Sider</sub>